unittests/platformagnostictests.py

```diff
@@ -421,12 +421,12 @@ def test_setup_with_unknown_option(self):
 
         with self.subTest('unknown user option'):
             out = self.init(testdir, extra_args=['-Dnot_an_option=1'], allow_fail=True)
-            self.assertIn('ERROR: Unknown options: "not_an_option"', out)
+            self.assertIn('ERROR: Unknown option: "not_an_option"', out)
 
         with self.subTest('unknown builtin option'):
             self.new_builddir()
             out = self.init(testdir, extra_args=['-Db_not_an_option=1'], allow_fail=True)
-            self.assertIn('ERROR: Unknown options: "b_not_an_option"', out)
+            self.assertIn('ERROR: Unknown option: "b_not_an_option"', out)
```

Comment:
I have the impression that these two hunks should have been included in the previous commit.

Reply:
The unknown option is now detected in `set_option` instead of `validate_cmd_line_options`, due to small differences between `accept_as_pending_option` and the now-removed loops in `initialize_from_top_level_project_call`. That changes the error message slightly.